frontend/packages/ux-editor/src/containers/DesignView/AddItem/useSearchComponent/useSearchComponent.ts

```diff
@@ -0,0 +1,71 @@
+import { useEffect, useState, type ChangeEvent, type KeyboardEvent } from 'react';
+import { useDebounce } from '@studio/hooks';
+import type { KeyValuePairs } from 'app-shared/types/KeyValuePairs';
+import type { IToolbarElement } from '../../../../types/global';
+
+const DEBOUNCE_DISABLED_TIME_MS = 1;
+const DEFAULT_DEBOUNCE_TIME_MS = 500;
+
+const getAllLabelWords = (toolbarItem: IToolbarElement, t: (key: string) => string): string[] => {
+  const translatedLabel = t('ux_editor.component_title.' + toolbarItem.type).toLowerCase();
+  const englishLabel = toolbarItem.label.toLowerCase();
+  return [...translatedLabel.split(/\s+/), ...englishLabel.split(/\s+/)];
+};
+
+const filterToolbarItems = (
+  items: IToolbarElement[],
+  search: string,
+  t: (key: string) => string,
+): IToolbarElement[] => {
+  if (!search) return items;
+  return items.filter((toolbarItem) =>
+    getAllLabelWords(toolbarItem, t).some((word) => word.startsWith(search)),
+  );
```

Comment:
Here we are doing two actions, maybe we can move the `.some` into another function to increase readability? 😄